integration-tests/bats/sql-merge.bats

```diff
@@ -1130,6 +1130,173 @@ SQL
+
+    run dolt sql -q "CALL DOLT_MERGE('merge_branch');"
+    log_status_eq 0
+
+    run dolt sql -q "select COUNT(*) from test"
+    [[ "$output" =~ 4 ]] || false
+
+    run dolt status
+    log_status_eq 0
+    [[ "$output" =~ "test2" ]] || false
+    [[ ! "$output" =~ "test1" ]] || false
+}
+
+@test "sql-merge: no-ff merge doesn't stomp working changes and doesn't fast forward" {
+    dolt commit -m "initial commit"
+    dolt checkout -b merge_branch
+    dolt SQL -q "INSERT INTO test values (9)"
+    dolt add test
+    dolt commit -m "modify test"
+
+    dolt checkout main
+    dolt SQL -q "CREATE TABLE t2 (pk int)"
+    dolt SQL -q "INSERT INTO t2 values (0)"
+    run dolt status
+    log_status_eq 0
+    [[ "$output" =~ "t2" ]] || false
+    [[ ! "$output" =~ "test" ]] || false
+
+    run dolt sql -q "CALL DOLT_MERGE('merge_branch', '--no-ff', '-m', 'no-ff merge');"
+    log_status_eq 0
+    [[ ! "$output" =~ "| 1            |" ]] || false
+
+    run dolt status
+    log_status_eq 0
+    [[ "$output" =~ "t2" ]] || false
+    [[ ! "$output" =~ "test" ]] || false
+
+    run dolt log
+    log_status_eq 0
+    [[ "$output" =~ "no-ff merge" ]] || false
+}
+
+@test "sql-merge: squash merge" {
+    dolt commit -m "initial commit"
+    dolt checkout -b merge_branch
+    dolt SQL -q "INSERT INTO test values (3)"
+    dolt add test
+    dolt commit -m "add pk 3 to test"
+
+    dolt checkout main
+    dolt SQL -q "INSERT INTO test values (4)"
+    dolt add test
+    dolt commit -m "add pk 4 to test1"
+
+    dolt SQL -q "CREATE TABLE t2 (pk int)"
+    dolt SQL -q "INSERT INTO t2 values (9)"
+    run dolt status
+    log_status_eq 0
+    [[ "$output" =~ "t2" ]] || false
+    [[ ! "$output" =~ "test1" ]] || false
+
+    run dolt sql -q "CALL DOLT_MERGE('--squash', 'merge_branch', '--no-commit');"
+    log_status_eq 0
+    [[ "$output" =~ "| 0            | 0         |" ]] || false
+
+    run dolt status
+    log_status_eq 0
+    [[ "$output" =~ "t2" ]] || false
+    [[ "$output" =~ "test" ]] || false
+
+    # make sure the squashed commit is not in the log.
+    dolt add .
+    dolt commit -m "squash merge"
+
+    run dolt log
+    log_status_eq 0
+    [[ "$output" =~ "add pk 4 to test" ]] || false
+    [[ ! "$output" =~ "add pk 3 to test" ]] || false
+}
+
+@test "sql-merge: --abort restores working changes" {
+    dolt commit -m "initial commit"
+    dolt branch other
+
+    dolt sql -q "INSERT INTO test VALUES (9),(8);"
+    dolt commit -am "added rows to test on main"
+
+    dolt checkout other
+    dolt sql -q "INSERT INTO test VALUES (7),(6);"
+    dolt commit -am "added rows to test on other"
+
+    dolt checkout main
+    # dirty the working set with changes to test2
+    dolt SQL -q "CREATE TABLE t2 (pk int)"
+    dolt sql -q "INSERT INTO t2 VALUES (5);"
+
+    dolt merge other --no-commit
+    dolt sql -q "call dolt_merge('--abort');"
+
+    run dolt sql -q "SELECT * from dolt_merge_status"
+    [[ "$output" =~ "false" ]] || false
```

Comment:
Put more of the table in there. "false" can show up in a lot of contexts, include stack traces - which is why you should verify the exit status as well.